### PR TITLE
Round values for relative time instead of flooring

### PR DESCRIPTION
**Pull request by gmta**
## Proposed change

Currently, the value used for displaying a relative time is floored before it's interpolated into the localized string. Many times, this has led me to misinterpret the actual value:

* 47 hours ago is displayed as "1 day ago" instead of "2 days ago"
* 13 days in the future is displayed as "in 1 week"

This change modifies the `relativeTime` function to use `Math.round` instead of `Math.floor` so the output more closely matches the actual relative time of the input.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Just put any entity with `device_class: timestamp` on a Lovelace dashboard and fake the state to try out different edge cases. Unfortunately, I could not get a simple template with `relative_time` to work.

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @GMTA, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#6225).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Review by balloob — APPROVED** [review]
Makes sense. Thanks !

**Comment by balloob** [comment]
Actually, we can't merge this unless there is also a PR to make sure the Python side is up to date: https://github.com/home-assistant/core/blob/dev/homeassistant/util/dt.py#L180 This version still uses rounding down.

**Comment by GMTA** [comment]
> Actually, we can't merge this unless there is also a PR to make sure the Python side is up to date

Makes sense, I'll look into it. Meanwhile I've updated this PR to get rid of the `floor` I erroneously introduced.

**Comment by GMTA** [comment]
@balloob I've also created a PR for core: https://github.com/home-assistant/core/pull/37125

**Comment by davet2001** [comment]
I have reviewed the HA core PR (looks good) 

But it doesn’t (and didn’t previously) output ‘weeks’ that were mentioned in the description above.

It must be that 13days=1week was coming from somewhere else as I think it would now output:

13days=>13 days
14days=>14 days
15days=>15 days
16days=>1 month

**Comment by GMTA** [comment]
@davet2001 The old code did send out "week" / "weeks", as can be seen here:

https://github.com/home-assistant/frontend/pull/6225/files#diff-94c31734a9546e7f680a38e3608c24eeL43

I've kept this behavior intact for the frontend:

https://github.com/home-assistant/frontend/pull/6225/files#diff-94c31734a9546e7f680a38e3608c24eeR24

This means that the new situation, just like the old one, means that frontend and core have different "fallback" time units of years and weeks, respectively.

**Pull request merged**
